core/app/[locale]/(default)/account/addresses/_actions/update-address.ts

```diff
@@ -0,0 +1,272 @@
+import { graphql, VariablesOf } from '~/client/graphql';
+import { TAGS } from '~/client/tags';
+import { FieldNameToFieldId } from '~/data-transformers/form-field-transformer/utils';
+
+import { type State } from './address-action';
+
+export const UpdateCustomerAddressMutation = graphql(`
+  mutation UpdateCustomerAddressMutation($input: UpdateCustomerAddressInput!) {
+    customer {
+      updateCustomerAddress(input: $input) {
+        errors {
+          __typename
+          ... on AddressDoesNotExistError {
+            message
+          }
+          ... on CustomerAddressUpdateError {
+            message
+          }
+          ... on CustomerNotLoggedInError {
+            message
+          }
+          ... on ValidationError {
+            message
+            path
+          }
+        }
+        address {
+          entityId
+        }
+      }
+    }
+  }
+`);
+
+const stringToNumber = z.coerce.string().pipe(z.coerce.number());
+
+const inputSchema = z.object({
+  addressEntityId: stringToNumber,
+  data: z.object({
+    firstName: z.string().optional(),
+    lastName: z.string().optional(),
+    address1: z.string().optional(),
+    address2: z.string().optional(),
+    city: z.string().optional(),
+    company: z.string().optional(),
+    countryCode: z.string().optional(),
+    stateOrProvince: z.string().optional(),
+    phone: z.string().optional(),
+    postalCode: z.string().optional(),
+    formFields: z
+      .object({
+        checkboxes: z.array(
+          z.object({
+            fieldEntityId: stringToNumber,
+            fieldValueEntityIds: z.array(stringToNumber),
+          }),
+        ),
+        multipleChoices: z.array(
+          z.object({
+            fieldEntityId: stringToNumber,
+            fieldValueEntityId: stringToNumber,
+          }),
+        ),
+        numbers: z.array(
+          z.object({
+            fieldEntityId: stringToNumber,
+            number: stringToNumber,
+          }),
+        ),
+        dates: z.array(
+          z.object({
+            fieldEntityId: stringToNumber,
+            date: z.string(),
+          }),
+        ),
+        passwords: z.array(
+          z.object({
+            fieldEntityId: stringToNumber,
+            password: z.string(),
+          }),
+        ),
+        multilineTexts: z.array(
+          z.object({
+            fieldEntityId: stringToNumber,
+            multilineText: z.string(),
+          }),
+        ),
+        texts: z.array(
+          z.object({
+            fieldEntityId: stringToNumber,
+            text: z.string(),
+          }),
+        ),
+      })
+      .optional(),
+  }),
+});
+
+function parseUpdateAddressInput(
+  value: Record<string, unknown>,
+  fields: Array<Field | FieldGroup<Field>>,
+): VariablesOf<typeof UpdateCustomerAddressMutation>['input'] {
+  const customFields = fields
+    .flatMap((f) => (Array.isArray(f) ? f : [f]))
+    .filter((field) => Boolean(value[field.name]))
```

Comment:
Difference between the add address function is that we filter out empty values here.